File: scripts/extract_transcript.py

import sqlite3
import os
import json

db_path = os.path.join(os.path.expanduser("~"), "AppData", "Roaming", "Ghost Writer", "ghost-writer.db")

if not os.path.exists(db_path):
    print(f"Database not found at: {db_path}")
    exit(1)

try:
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    # Find meeting ID
    cursor.execute("SELECT id FROM meetings WHERE title LIKE '%Web Form Creation Training%'")
    meeting = cursor.fetchone()
    
    if not meeting:
        print("Meeting not found")
        exit(1)
        
    meeting_id = meeting[0]
    
    # Get transcript
    cursor.execute("SELECT speaker, content FROM transcripts WHERE meeting_id = ? ORDER BY timestamp_ms ASC", (meeting_id,))
    rows = cursor.fetchall()
    
    transcript = [{"speaker": r[0], "text": r[1]} for r in rows]
    print(json.dumps(transcript))
    
    conn.close()
except Exception as e:
    print(f"Error: {e}")
    exit(1)

File: scripts/extract_transcript.js

const Database = require('better-sqlite3');
const path = require('path');
const os = require('os');
const fs = require('fs');

// Path to Ghost Writer DB on Windows
const dbPath = path.join(os.homedir(), 'AppData', 'Roaming', 'Ghost Writer', 'ghost-writer.db');

if (!fs.existsSync(dbPath)) {
    console.error('Database not found at:', dbPath);
    process.exit(1);
}

const db = new Database(dbPath);

try {
    const meeting = db.prepare("SELECT id FROM meetings WHERE title LIKE '%Web Form Creation Training%'").get();
    
    if (!meeting) {
        console.error('Meeting not found');
        process.exit(1);
    }

    const transcript = db.prepare("SELECT speaker, content FROM transcripts WHERE meeting_id = ? ORDER BY timestamp_ms ASC")
        .all(meeting.id);

    console.log(JSON.stringify(transcript, null, 2));
} catch (err) {
    console.error('Error querying database:', err);
    process.exit(1);
}
